### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by compnerd**
Use `PROJECT_SOURCE_DIR` to allow embedding within larger projects. `CMAKE_SOURCE_DIR` is the root of the source tree, where as `PROJECT_SOURCE_DIR` is the directory of the current project.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @stephentyrone

**Review by dabrahams — APPROVED** [review]
This is a good change

**Comment by dabrahams** [comment]
@stephentyrone bump

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ cmake_minimum_required(VERSION 3.16)
 project(swift-numerics
   LANGUAGES Swift)
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
```
[the remark this answers is not in the document]
Comment (compnerd) [reply CMakeLists.txt:14]: Yes, I think that this is safe because of that.

**Review by stephentyrone — APPROVED** [review]

**Pull request merged**